IonicPortals/IonicPortals/Portal.swift

```diff
@@ -1,19 +1,87 @@
 import Foundation
+import Capacitor
 import IonicLiveUpdates
 
-@objc(Portal)
-public class Portal: NSObject {
+/// The configuration of a web application to be embedded in an iOS application
+public struct Portal {
     
-    // MARK: - Instance Properties
-    public let name: String;
-    public var initialContext: Dictionary<String, Any>?
-    public var startDir: String?
+    /// The name of the portal
+    public let name: String
     
-    public var liveUpdateConfig: LiveUpdate? = nil
+    /// The root directory of the ``Portal`` relative to root of the `Bundle`
+    public let startDir: String
     
-    public init(_ name: String, _ startDir: String?) {
+    /// Any initial state required by the web application
+    public var initialContext: [String: JSValue]
+    
+    /// The `LiveUpdate` configuration used to determine the location of updated application assets.
+    public var liveUpdateConfig: LiveUpdate? = nil {
+        didSet {
+            guard let liveUpdateConfig = liveUpdateConfig else { return }
+            try? LiveUpdateManager.shared.add(liveUpdateConfig)
+        }
+    }
+    
+    /// Creates an instance of ``Portal``
+    /// - Parameters:
+    ///   - name: The name of the portal, must be unique.
+    ///   - startDir: The starting directory of the ``Portal`` relative to the root of the ``Bundle``.
+    ///     If `nil`, the portal name is used as the starting directory. Defaults to `nil`.
+    ///   - initialContext: Any initial state rqeuired by the web application. Defaults to `[:]`.
+    ///   - liveUpdateConfig: The `LiveUpdate` configuration used to determine to location of updated application assets. Defaults to `nil`.
+    public init(name: String, startDir: String? = nil, initialContext: [String: JSValue] = [:], liveUpdateConfig: LiveUpdate? = nil) {
         self.name = name
         self.startDir = startDir ?? name
+        self.initialContext = initialContext
+        self.liveUpdateConfig = liveUpdateConfig
+        if let liveUpdateConfig = liveUpdateConfig {
+            try? LiveUpdateManager.shared.add(liveUpdateConfig)
+        }
     }
 }
 
+/// The objective-c representation of ``Portal``
+@objc public class IONPortal: NSObject {
+    internal var portal: Portal
+    
+    /// The name of the portal
+    @objc public var name: String { portal.name }
+    
+    /// The root directory of the ``Portal`` relative to root of the `Bundle`
+    @objc public var startDir: String { portal.startDir }
+    
+    /// Any initial state required by the web application
+    @objc public var initialContext: [String: Any] {
```

Comment:
Should probably add documentation on what types are valid here. This was one of the factors that convinced me to go this route, because `JSValue` is a Swift protocol and can't be expressed in objective-c.